proposesolution/ai.go

```diff
@@ -0,0 +1,52 @@
+package proposesolution
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/generative-ai-go/genai"
+	"google.golang.org/api/option"
+
+	"github.com/securego/gosec/v2/issue"
+)
+
+const (
+	GeminiModel    = "gemini-1.5-flash"
+	AIPrompt       = "What is the solution to fix the error \"%s\". Answer limited to 200 words"
+	GeminiProvider = "gemini"
+)
+
+func generateSolutionByGemini(aiApiKey string, issues []*issue.Issue) error {
+	timeout := 30 * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	client, err := genai.NewClient(ctx, option.WithAPIKey(aiApiKey))
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	model := client.GenerativeModel(GeminiModel)
+	for _, issue := range issues {
+		prompt := fmt.Sprintf(AIPrompt, issue.What)
+		resp, err := model.GenerateContent(ctx, genai.Text(prompt))
```

Comment:
What is the performance of this repeated call when there are many issues in a report? It will be an call for each issue. Is there a way to beach multiple issues in one call to improve the performance? Maybe some caching would be also helpful to avoid calling multiple times for the same issue type.